src/senaite/core/api/geo.py

```diff
@@ -199,3 +202,91 @@ def get_country_or_subdivision(thing, default=_marker):
         raise ValueError("Could not find a record for '{}'".format(
             thing.lower()))
     return default
+
+
+def to_dms(degrees, precision=4, default=_marker):
+    """Converts a geographical coordinate in decimal degrees to a dict with
+    degrees, minutes and seconds as the keys
+
+    :param degrees: coordinate in decimal degrees
+    :type degrees: string,int,float
+    :param precision: number of decimals for seconds
+    :type precision: int
+    :return: a dict with degrees, minutes and seconds as keys
+    """
+    if not is_floatable(degrees):
+        if default is _marker:
+            raise ValueError("Decimal degree is not valid")
```

Comment:
This should be better a `TypeError`, because it is at this point a type that is not floatable. A more helpful error message could then be `Expected degrees to be a floatable decimal, but got the type '%r' instead` % type(degrees)

Reply:
Not sure, note the type does not need to be numeric, but the *value* has to be floatable. E.g., `"2"` is from `str` type, but floatable

Reply:
This https://github.com/senaite/senaite.core/pull/2565/commits/f43c767949387d498e41ee5698b264d33433fb7b fixes the message though